platform/saivpp/vpplib/vppxlate/SaiVppXlate.c

```diff
@@ -1828,6 +1890,148 @@ int vpp_acl_del (uint32_t acl_index)
     return ret;
 }
 
+int vpp_tunterm_acl_interface_add_del (uint32_t tunterm_index, bool is_bind, const char *hwif_name)
+{
+    vat_main_t *vam = &vat_main;
+    vl_api_tunterm_acl_interface_add_del_t *mp;
+    int ret;
+
+    VPP_LOCK();
+
+    __plugin_msg_base = tunterm_msg_id_base;
+    M (TUNTERM_ACL_INTERFACE_ADD_DEL, mp);
+
+    if (hwif_name) {
+        u32 idx;
+
+        idx = get_swif_idx(vam, hwif_name);
+        if (idx != (u32) -1) {
+            mp->sw_if_index = htonl(idx);
+        } else {
+            SAIVPP_ERROR("Unable to get sw_index for %s\n", hwif_name);
+            VPP_UNLOCK();
+            return -EINVAL;
+        }
+    } else {
+        VPP_UNLOCK();
+        return -EINVAL;
+    }
+    mp->is_add = is_bind;
+    mp->tunterm_acl_index= htonl(tunterm_index);
+
+    S (mp);
+    W (ret);
+
+    VPP_UNLOCK();
+
+    return ret;
+}
+
+
+int vpp_tunterm_acl_del (uint32_t tunterm_index)
+{
+    vat_main_t *vam = &vat_main;
+    vl_api_tunterm_acl_del_t *mp;
+    int ret;
+
+    VPP_LOCK();
+
+    __plugin_msg_base = tunterm_msg_id_base;
+    M (TUNTERM_ACL_DEL, mp);
+
+    mp->tunterm_acl_index= htonl(tunterm_index);
+
+    S (mp);
+    W (ret);
+
+    VPP_UNLOCK();
+
+    return ret;
+}
+
+int vpp_tunterm_acl_add_replace (uint32_t *tunterm_index, uint32_t count, vpp_tunerm_acl_t *in_acl)
+{
+    u32 idx;
+    vat_main_t *vam = &vat_main;
+    vpp_ip_addr_t *addr;
+    tunterm_acl_rule_t *in_rule;
+    vl_api_address_t *api_addr;
+    vl_api_tunterm_acl_rule_t *vpp_rule;
+    vl_api_tunterm_acl_add_replace_t *mp;
+    int ret;
+
+    init_vpp_client();
+
+    VPP_LOCK();
+
+    __plugin_msg_base = tunterm_msg_id_base;
+
+    M2 (TUNTERM_ACL_ADD_REPLACE, mp, count*sizeof(vl_api_tunterm_acl_rule_t));
+
+    mp->count = htonl(count);
+    mp->tunterm_acl_index = htonl(*tunterm_index);
+
+    for (idx = 0; idx < count; idx++) {
+        in_rule = &in_acl->rules[idx];
+        vpp_rule = &mp->r[idx];
+
+        addr = &in_rule->dst_prefix;
+        api_addr = &vpp_rule->dst;
+
+        if (addr->sa_family == AF_INET) {
```

Comment:
Here too you can use vpp_to_vl_api_ip_addr() to convert IP addresses

Reply:
Updated